mathesar/tests/api/test_column_api.py

```diff
@@ -494,7 +494,7 @@ def test_column_update_type_dynamic_default(column_test_table, client):
     response = client.patch(
         f"/api/db/v0/tables/{column_test_table.id}/columns/{column.id}/", data=data
     )
-    assert response.status_code == 400
+    assert response.status_code == 200
```

Comment:
We didn't allow this before since we didn't have a way to construct `cast_to_new_type(adefault_generating_func())` in python. This is because SQLAlchemy receives _all_ default values as simple strings, and it's rather tricky to then reconstruct the underlying function call in the case of a dynamic default value.

This issue is negated when doing the work on the DB.

Reply:
Ah, so we're now, when casting, also redefinining the dynamic default to cast itself to the new type, thus not actually losing the dynamic default? That would set up an interesting "entropy" collection scheme, where if you repeatedly cast the dynamic default between a few types you lose information, not that that's a problem.